relay-server/src/services/project.rs

```diff
@@ -579,52 +579,72 @@ impl Project {
         }
     }
 
-    /// Remove metric buckets that are not allowed to be ingested.
+    /// Remove metric buckets or tags that are not allowed to be ingested.
     fn filter_metrics(&self, metrics: &mut Vec<Bucket>) {
         let Some(state) = &self.state_value() else {
             return;
         };
 
-        Self::apply_metrics_deny_list(&state.config.metrics, metrics);
-        Self::filter_disabled_namespace(state, metrics);
-    }
-
-    fn apply_metrics_deny_list(deny_list: &ErrorBoundary<Metrics>, buckets: &mut Vec<Bucket>) {
-        let ErrorBoundary::Ok(metrics) = deny_list else {
-            return;
-        };
+        let metrics_config = &state.config.metrics;
 
-        buckets.retain(|bucket| {
-            if metrics.denied_names.is_match(&bucket.name) {
+        metrics.retain_mut(|bucket| {
+            if Self::metric_name_denied(metrics_config, bucket) {
                 relay_log::trace!(mri = bucket.name, "dropping metrics due to block list");
                 false
+            } else if Self::metric_namespace_invalid(state, bucket) {
```

Comment:
`metric_namespace_invalid` tracks trace logs internally when it drops the bucket, but `metric_name_denied` does not. It's easy to miss this difference, so could we align this?

This function actually drops due to two reasons: an MRI that cannot be parsed, or an unknown namespace. If you prefer to have this in sub functions, I'd suggest to split those into two functions that just return booleans, and then log the messages here at the upper level. This would require you to pass the parsed MRI into the namespace function instead of the entire bucket.

Reply:
agree thats better! I moved it up to the upper function now.